src/TraceEvent/TraceEvent.cs

```diff
@@ -1869,6 +1872,11 @@ protected internal virtual void Validate()
         [Conditional("DEBUG")]
         protected internal void DebugValidate()
         {
+            // TODO FIX NOW For EventPipe data, the validation of EventDataLength for the following data always fail. 
+            // GCSuspendEETraceData, GCNoUserDataTraceData, MethodLoadUnloadVerboseTraceData, MethodJittingStartedTraceData
+            if (this.source.origin == EventPipeEventSource.EventPipe)
```

Comment:
I re-enable the validation. I can work with you to check why the EventDataLength is wrong.